config/application.rb

```diff
@@ -29,5 +29,9 @@ class Application < Rails::Application
     config.time_zone = 'London'
 
     config.active_record.raise_in_transactional_callbacks = true
+
+    config.action_dispatch.rescue_responses.merge!(
+      'StateMachines::InvalidTransition' => :unprocessable_entity
```

Comment:
This is the magic line: it causes `InvalidTransition` exceptions to be turned into 422 errors. (Except in development mode, *unless* you disable the `config.consider_all_requests_local = true` option in `development.rb`). I think this will always be the right thing to do: it's the same as `ActiveRecord::RecordInvalid`, for example.